### PR TITLE
[RHCLOUD-20896] Moving binder.go into new package and updating tests

### PR DESCRIPTION
**Pull request by Ellen-Yi-Dong**
Moved binder.go, binder_test.go, echo_context.go, and echo_context_test.go into new echo package within util package.

Deleted echo.Binder() within the different test files. 

Made sure that the files imported the correct files. 

Within some of the handlers.go files, after the Bind() function, made sure to change util.NewBadRequestErr(err) to err. 

JIRA ticket: [RHCLOUD-20896](https://issues.redhat.com/browse/RHCLOUD-20896)

### CODE REVIEW

**Comment by Ellen-Yi-Dong** [comment]
#523

**Comment by Ellen-Yi-Dong** [comment]
#523 (the changes from this ticket is within this PR 554)

File: dao/request_params.go [hunk 1]
```diff
@@ -14,12 +13,12 @@ type RequestParams struct {
 }
 
 func NewRequestParamsFromContext(c echo.Context) (*RequestParams, error) {
-	tenantId, err := util.GetTenantFromEchoContext(c)
+	tenantId, err := sourcesEcho.GetTenantFromEchoContext(c)
```
Comment (lindgrenj6) [inline dao/request_params.go]: What would you think of renaming this import `echoUtils`? Since it's the echo-related util functions just in their own package.

**Review by lindgrenj6 — COMMENTED** [review]
Just one tiny comment, otherwise looks good!

**Review by petracihalova — APPROVED** [review]
nice job !!!

**Review by MikelAlejoBR — APPROVED** [review]
Nice one!

File: binder_test.go [hunk 2]
```diff
@@ -1,121 +0,0 @@
...
-
-// TestEmptyJsonBody tests that when a valid, empty JSON object is given, a "bad request" error is returned.
-func TestEmptyJsonBody(t *testing.T) {
-	testValues := []*bytes.Buffer{
-		bytes.NewBufferString("{}"),
-		bytes.NewBufferString("{     }"),
-		bytes.NewBufferString("[]"),
-		bytes.NewBufferString("[     ]"),
-		bytes.NewBufferString("null"),
-	}
```
Comment (petracihalova) [inline binder_test.go]: this test is removed ... I think we still need a test for empty body in the request

**Review by lindgrenj6 — APPROVED** [review]
